src/commands/showReferrer.ts

```diff
@@ -0,0 +1,63 @@
+import * as vscode from 'vscode';
+import { execAsync } from '../support/execAsync';
+import { window } from 'vscode';
+import * as os from 'os';
+import * as path from 'path';
+import { access } from 'fs/promises'; //this import accepts one argument for calling access
+
+export async function showReferrer(imageTag: any): Promise<void> {
+    try {
+        async function getOrasPath() {
+            // check oras exist on user computer
+            const isWindows = os.platform() === 'win32'; //identifying if its a windows platform using os.plateform, if it is then isWindows will be true
+            const homedir = os.homedir(); //getting the host computer home directory
+            const orasExecutable = isWindows ? 'oras.exe' : 'oras'; //switches use of executable files. if true use oras.exe and if false use oras
+            const dirOras = path.join(homedir, 'bin', 'oras', orasExecutable); //returns the path to oras
+
+            try {
+                await access(dirOras); // Check if the file exists, this will throw an error if the file doesn't exist
+                return dirOras; // If the file exists, return the dirOras path
+            } catch (err) {
+                 // Throw an error with the message
+                const errorMessage = 'ORAS not found on host computer. Download ORAS: ';
+                const goToOras = 'Download ORAS';
```

Comment:
Is this a link?  If so, should the variable name be 'goToOrasLink'?

Reply:
This isn't the link itself but the naming of the button that when selected will execute the link.

Reply:
the variable name should describe what it is, ie. 'goToOrasButton'